### PR TITLE
Arousal valence models - Regression tasks

### PR DESCRIPTION
**Pull request by xaviliz**
Added functionalities for regression tasks.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
 essentia  # TODO: Fix Essentia version once we have a wheel with the required pytools
 joblib~=0.13.2  # also a dependency of pescador, scikit-learn
-numpy~=1.16.4 # also a dependency of essentia, essentia-tensorflow, tensorflow, scikit-image, scikit-learn, scipy
```
Comment (palonso) [inline requirements.txt]: Let's discard version changes from this PR.
We can address a version upgrade considering the base repository once we merge the new functionalities.

File: src/score_predictions.py [hunk 2]
```diff
@@ -54,87 +54,195 @@ def score_predictions(args):
...
-
-    y_true = np.array(y_true)
-    y_pred = np.array(y_pred)
-    if shared.type_of_groundtruth(y_true) == "multilabel-indicator":
-        y_pred_indicator = np.round(y_pred)
-        report = classification_report(y_true, y_pred_indicator)
+def get_metrics(y_true, y_pred, fold_gt, fold_pred, n_folds, task_type):
+
+    if task_type == "regression":
+        # shared.type_of_groundtruth(y_true) == "continuous-multioutput":    # regression
```
Comment (palonso) [inline src/score_predictions.py]: deprecated code?

File: src/score_predictions.py [hunk 3]
```diff
@@ -54,87 +54,195 @@ def score_predictions(args):
...
+
+        # compute micro metrics
+        micro = {}
```
Comment (palonso) [inline src/score_predictions.py]: micro -> micro_metrics

File: src/score_predictions.py [hunk 4]
```diff
@@ -54,87 +54,195 @@ def score_predictions(args):
...
 
 
 def store_results(
     output_file,
-    roc_auc_score,
-    pr_auc_score,
-    macro_acc_score,
-    micro_acc,
+    scores,
+    micro,
```
Comment (palonso) [inline src/score_predictions.py]: Why not packing `micro` into scores too?

**Review by palonso — CHANGES_REQUESTED** [review]
Thanks for the PR @xaviliz 
I requested some minor changes

File: requirements.txt (hunk 1 above)
Comment (xaviliz) [reply requirements.txt]: You welcome! Sure, no problem. These were some changes we did together. There was issues with versions when we tried to install the packages. I will keep it in my machine to update it later.

File: src/score_predictions.py (hunk 2 above)
Comment (xaviliz) [reply src/score_predictions.py]: right!

File: src/score_predictions.py (hunk 4 above)
Comment (xaviliz) [reply src/score_predictions.py]: ok, done!

**Comment by palonso** [comment]
Thanks for the changes @xaviliz 

Before merging, we need to update `get_metrics` in the tests.

https://github.com/palonso/musicnn-training/blob/d8ed47db82e9c966c069d844a372b61f25730aa0/test/test_score_predictions.py#L45

Also it would be great to add a new test for the regression metrics using `test_get_metrics` as a template.

**Comment by palonso** [comment]
I see that there is an issue with sklearn. The [version](https://scikit-learn.org/0.21/modules/generated/sklearn.metrics.mean_squared_error.html) we use for testing doesn't support the `squared` parameter.
I have tested the branch locally and I found that with `scikit-learn==1.0.2` the dependencies can be installed correctly on Py3.7 and all the tests pass. Let's update the requirements to `scikit-learn~=1.0.2`.

Also, the tests were meant to be run in github actions with pytest (you can also run these commands locally for testing):
https://github.com/palonso/musicnn-training/blob/d8ed47db82e9c966c069d844a372b61f25730aa0/.github/workflows/python-ci.yaml#L38
so the unittest wrapper is not actually required.

I think the eassiest is to go back the last 4 commits (`git reset --soft HEAD~4`), and then `push --force` a new commit adding the `test_get_metrics_for_classification` function exclusively.

**Comment by xaviliz** [comment]
thanks @palonso 
I will let you know when it is fixed

**Comment by xaviliz** [comment]
Alright, it is done.

**Pull request merged**